pandas-stubs/io/parsers/readers.pyi

```diff
@@ -0,0 +1,438 @@
+    encoding: str | None = ...,
+    encoding_errors: str | None = ...,
+    dialect: str | csv.Dialect = ...,
+    on_bad_lines: Literal["error", "warn", "skip"]
+    | Callable[[list[str]], list[str] | None] = ...,
+    delim_whitespace: bool = ...,
+    low_memory: bool = ...,
+    memory_map: bool = ...,
+    float_precision: Literal["high", "legacy", "round_trip"] | None = ...,
+    storage_options: StorageOptions | None = ...,
+) -> TextFileReader: ...
+@overload
+def read_table(
+    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
+    *,
+    sep: str | None | lib.NoDefault = ...,
+    delimiter: str | None | lib.NoDefault = ...,
+    header: int | Sequence[int] | Literal["infer"] | None = ...,
+    names: list[str] | None | lib.NoDefault = ...,
+    index_col: int | str | Sequence[str | int] | Literal[False] | None = ...,
+    usecols: list[str]
+    | tuple[str, ...]
+    | Sequence[int]
+    | Series
+    | Index
+    | npt.NDArray
+    | Callable[[str], bool]
+    | None = ...,
+    dtype: DtypeArg | None = ...,
+    engine: CSVEngine | None = ...,
+    converters: dict[int | str, Callable[[str], Any]] = ...,
+    true_values: list[str] = ...,
+    false_values: list[str] = ...,
+    skipinitialspace: bool = ...,
+    skiprows: int | Sequence[int] | Callable[[int], bool] = ...,
+    skipfooter: int = ...,
+    nrows: int | None = ...,
+    na_values: Sequence[str] | dict[str, Sequence[str]] = ...,
+    keep_default_na: bool = ...,
+    na_filter: bool = ...,
+    verbose: bool = ...,
+    skip_blank_lines: bool = ...,
+    parse_dates: bool
+    | Sequence[int]
+    | list[str]
+    | Sequence[Sequence[int]]
+    | dict[str, Sequence[int]] = ...,
+    infer_datetime_format: bool = ...,
+    keep_date_col: bool = ...,
+    date_parser: Callable = ...,
+    dayfirst: bool = ...,
+    cache_dates: bool = ...,
+    iterator: Literal[False] = ...,
+    chunksize: None = ...,
+    compression: CompressionOptions = ...,
+    thousands: str | None = ...,
+    decimal: str = ...,
+    lineterminator: str | None = ...,
+    quotechar: str = ...,
+    quoting: int = ...,
+    doublequote: bool = ...,
+    escapechar: str | None = ...,
+    comment: str | None = ...,
+    encoding: str | None = ...,
+    encoding_errors: str | None = ...,
+    dialect: str | csv.Dialect = ...,
+    on_bad_lines: Literal["error", "warn", "skip"]
+    | Callable[[list[str]], list[str] | None] = ...,
+    delim_whitespace: bool = ...,
+    low_memory: bool = ...,
+    memory_map: bool = ...,
+    float_precision: Literal["high", "legacy", "round_trip"] | None = ...,
+    storage_options: StorageOptions | None = ...,
+) -> DataFrame: ...
+def read_fwf(
+    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
+    colspecs: Sequence[tuple[int, int]] | str | None = ...,
+    widths: Sequence[int] | None = ...,
+    infer_nrows: int = ...,
+    **kwds: Any,
+) -> DataFrame | TextFileReader: ...
```

Comment:
I am hazarding a guess here, but I'm guessing if you pass `chunksize` and/or `iterator` then it returns `TextFileReader` and if you don't, you get `DataFrame`.

So if `chunksize` and/or `iterator` are one of the `kwargs` that can be passed to `TextFileReader`, we might as well do the overload with those specific arguments as is done for `read_csv()` and `read_table()`.